core/src/main/java/org/infinispan/xsite/statetransfer/XSiteStateTransferManagerImpl.java

```diff
@@ -365,14 +367,14 @@ private BackupRpcConfiguration backupRpcConfiguration(String siteName) {
 
    private Map<Address, Response> invokeRemotelyInLocalSite(CacheRpcCommand command) throws Exception {
       commandsFactory.initializeReplicableCommand(command, false);
-      CompletableFuture<Map<Address, Response>> remoteFuture = rpcManager.invokeRemotelyAsync(null,
-            command, rpcManager.getDefaultRpcOptions(true, DeliverOrder.NONE));
+      RpcOptions options = rpcManager.getDefaultRpcOptions(true, DeliverOrder.NONE);
+      CompletableFuture<Map<Address, Response>> remoteFuture = rpcManager.invokeRemotelyAsync(null, command, options);
       final Future<Response> localFuture = asyncExecutor.submit(
             LocalInvocation.newInstance(responseGenerator, command, commandsFactory, rpcManager.getAddress()));
       final Map<Address, Response> responseMap = new HashMap<>();
       responseMap.put(rpcManager.getAddress(), localFuture.get());
       //noinspection unchecked
-      responseMap.putAll(remoteFuture.get());
+      responseMap.putAll(RpcUtil.waitFor(remoteFuture, options.timeout(), options.timeUnit()));
```

Comment:
Could you make this async and send the reply from `remoteFuture.whenCompleted()` instead?